include/cetl/pmr/buffer_memory_resource_delegate.hpp

```diff
@@ -0,0 +1,266 @@
+    reallocate_internal(UpstreamResourceType& upstream,
+                        void*                 data,
+                        std::size_t           old_object_count,
+                        std::size_t           new_object_count,
+                        std::size_t           new_align)
+    {
+        (void) old_object_count;
+        void* result = nullptr;
+        if (data == in_use_)
+        {
+            result  = allocate_internal_buffer(new_object_count, new_align);
+            in_use_ = result;
+        }
+        else
+        {
+            result = upstream.reallocate(data, old_object_count, new_object_count, new_align);
+        }
+        return result;
+    }
+
+public:
+    /// Designated constructor that initializes the object with a fixed buffer and an optional upstream memory resource.
+    /// @param buffer                   The buffer that is used to satisfy allocation requests.
+    /// @param buffer_size_bytes        The size, in bytes, of the buffer.
+    /// @param upstream                 An optional upstream memory resource to use if the buffer is already in use.
+    /// @param upstream_max_size_bytes  The maximum size of the upstream buffer.
+    UnsynchronizedBufferMemoryResourceDelegate(void*                       buffer,
+                                               std::size_t                 buffer_size_bytes,
+                                               UpstreamMemoryResourceType* upstream,
+                                               std::size_t                 upstream_max_size_bytes) noexcept
+        : upstream_{upstream}
+        , buffer_{buffer}
+        , buffer_size_bytes_{buffer_size_bytes}
+        , max_size_bytes_{calculate_max_size_bytes(buffer_size_bytes, upstream_max_size_bytes)}
+        , upstream_max_size_bytes_{upstream_max_size_bytes}
+        , in_use_{nullptr}
+    {
+        CETL_DEBUG_ASSERT(nullptr != upstream,
+                          "CDE_ubmrd_001: Upstream memory resource cannot be null. Use std::pmr::null_memory_resource "
+                          "or cetl::pmr::null_memory_resource if you don't want an upstream memory resource.");
+        CETL_DEBUG_ASSERT(nullptr != buffer || buffer_size_bytes == 0,
+                          "CDE_ubmrd_002: Nullptr buffer provided with a buffer size > 0.");
+    }
+
+    ~UnsynchronizedBufferMemoryResourceDelegate()                                                            = default;
+    UnsynchronizedBufferMemoryResourceDelegate(const UnsynchronizedBufferMemoryResourceDelegate&)            = delete;
+    UnsynchronizedBufferMemoryResourceDelegate& operator=(const UnsynchronizedBufferMemoryResourceDelegate&) = delete;
+    UnsynchronizedBufferMemoryResourceDelegate(UnsynchronizedBufferMemoryResourceDelegate&&) noexcept        = default;
+    UnsynchronizedBufferMemoryResourceDelegate& operator=(UnsynchronizedBufferMemoryResourceDelegate&&)      = delete;
```

Comment:
Again, why the move constructor but not the move operator?

Reply:
Same answer as above.